modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/CommunicationTcpUtils.java

```diff
@@ -201,4 +205,161 @@ public static void failNode(ClusterNode nodeToFail,
             ", err=" + err +
             ", connectErrs=" + X.getSuppressedList(err) + ']');
     }
+
+    /**
+     * Base class for network interface matchers.
+     */
+    public abstract static class NetworkInterfaceMatcher {
```

Comment:
Why are these types internal to `CommunicationTcpUtils`?

Reply:
Moved to separate files